packages/vite/src/hbs.ts

```diff
@@ -14,9 +14,17 @@ export function hbs(): Plugin {
   return {
     name: 'rollup-hbs-plugin',
     enforce: 'pre',
-    async resolveId(source: string, importer: string | undefined) {
+    async resolveId(source: string, importer: string | undefined, options) {
+      if (options.custom?.embroider?.isExtensionSearch) {
+        return null;
+      }
       let resolution = await this.resolve(source, importer, {
         skipSelf: true,
+        custom: {
```

Comment:
I think in the discussion we agreed to only add this when you're actually doing an extension search, so in the resolve inside `maybeSynthesizeComponentJS` on line 87